src/main/java/seedu/address/model/person/Vendor.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Vendor in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Vendor extends Person {
+
+
+    /**
+     * Every field must be present and not null.
```

Comment:
Per our user guide, only name ought to be compulsory for Vendor.

Reply:
According to this implementation, all fields are necessary and the user cannot leave them empty. Since `Email` and `Address` are not relevant to our project, the only issue is with `Phone`. To follow LSP, perhaps we can leave this be until we adapt `Person` itself?